### PR TITLE
Add docs for secrets based on existing credentials

### LINKED ISSUE #314: Update Chains authentication doc with auth for DockerHub
# Expected Behavior
Chains is able to sign images on Docker Hub.

# Actual Behavior
Chains is unable to sign images on Docker Hub while it works fine on GCR.

# Steps to Reproduce the Problem

I followed the [signed provenance tutorial](https://github.com/tektoncd/chains/blob/main/docs/tutorials/signed-provenance-tutorial.md) so these steps are from there:

1.  Install Tekton Chains
1. Set up authentication for the registry (namespace `default`) -
    ```
    kubectl create secret docker-registry registry-credentials --docker-server=docker.io --docker-username=$DOCKERHUB_USERNAME --docker-email=$DOCKERHUB_EMAIL --docker-password=$DOCKERHUB_PASSWORD`
    kubectl patch serviceaccount default  -p "{\"imagePullSecrets\": [{\"name\": \"registry-credentials\"}]}" -n default
    ```
1. Generate a key pair
    ```
    cosign generate-key-pair k8s://tekton-chains/signing-secrets
    ```
1. Configure authentication for the kaniko task:
    ```
    docker login (log in to Docker Hub)
    kubectl create secret generic secret-docker-hub --from-file ~/.docker/config.json
    ```
1. Configure Tekton Chains
    ```
    kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"artifacts.taskrun.format": "in-toto"}}'
    kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"artifacts.taskrun.storage": "oci"}}'
    kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"transparency.enabled": "true"}}'
    ```
1. Apply and start the `kaniko` task
    ```
    kubectl apply -f https://raw.githubusercontent.com/tektoncd/chains/main/examples/kaniko/kaniko.yaml
    tkn task start --param IMAGE=docker.io/<Docker Hub registry name>/kaniko-chains --use-param-defaults --workspace name=source,emptyDir="" --workspace name=dockerconfig,secret=secret-docker-hub kaniko-chains
    ```
1. Wait for the TaskRun to finish and for it to report that signing has failed
    ```
    kubectl get tr [TASKRUN_NAME] -o json | jq -r .metadata.annotations
    ```
    Look for, `"chains.tekton.dev/signed": "failed"` in the annotations

# Additional Info

- Here is a relevant error from the `tekton-chains-controller` pod:
`
{"level":"error","ts":"2022-01-01T13:39:18.427Z","logger":"watcher","caller":"chains/signing.go:210","msg":"POST https://index.docker.io/v2/concaf/kaniko-chains/blobs/uploads/: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:concaf/kaniko-chains Type:repository] map[Action:push Class: Name:concaf/kaniko-chains Type:repository]]","commit":"0c28d26","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"8b965ddc-4fd9-44db-9763-6ade25dd06c1","knative.dev/key":"default/kaniko-chains-run-k6wlb","stacktrace":"github.com/tektoncd/chains/pkg/chains.(*TaskRunSigner).SignTaskRun\n\tgithub.com/tektoncd/chains/pkg/chains/signing.go:210\ngithub.com/tektoncd/chains/pkg/reconciler/taskrun.(*Reconciler).FinalizeKind\n\tgithub.com/tektoncd/chains/pkg/reconciler/taskrun/taskrun.go:61\ngithub.com/tektoncd/chains/pkg/reconciler/taskrun.(*Reconciler).ReconcileKind\n\tgithub.com/tektoncd/chains/pkg/reconciler/taskrun/taskrun.go:42\ngithub.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/taskrun.(*reconcilerImpl).Reconcile\n\tgithub.com/tektoncd/pipeline@v0.27.1-0.20210830150214-8afd1563782d/pkg/client/injection/reconciler/pipeline/v1beta1/taskrun/reconciler.go:249\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\tknative.dev/pkg@v0.0.0-20211123135150-787aec59e70a/controller/controller.go:542\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\tknative.dev/pkg@v0.0.0-20211123135150-787aec59e70a/controller/controller.go:478"}
`
- I think the code that is failing is here - https://github.com/tektoncd/chains/blob/243e90b3736f2fa4daa013c38c1cefba254cf106/pkg/chains/storage/oci/oci.go#L188-L191
but I was not able to dig deeper since I don't understand it fully.
- I suspect this might be related to https://github.com/sigstore/cosign/issues/587
- Kubernetes version:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-21T23:01:33Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```

- Tekton Pipeline version:
```
$ tkn version
Client version: dev
Pipeline version: v0.31.0
```

- Docker version - `20.10.12`

### PR DESCRIPTION
**Pull request by concaf**
The current docs on "Authenticating to an OCI Registry" cover creating a
secret by providing credentials on the command line but do not cover
creating a secret based on existing credentials; this commit adds that.

Fixes #314 

CC: @priyawadhwa

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -18,16 +18,31 @@ This doc will cover how to set this up!
...
 Set the namespace and name of the Kubernetes service account:
 
 ```shell
 export NAMESPACE=<your namespace>
 export SERVICE_ACCOUNT_NAME=<service account name>
 ```
 
+### Create a Secret based on existing credentials
+
+If you already ran `docker login`, you can copy the credentials into Kubernetes:
```
Comment (priyawadhwa) [inline docs/authentication.md:30]: Can we add a step here where users confirm that the ~/.docker/config.json file is in this format?

```
{
  "auths": {
    "https://index.docker.io/v1/": {
      "auth": "redacted"
    }
}
```

on mac the dockerconfig uses keychain so it usually isn't in this format

Comment (concaf) [reply docs/authentication.md]: @priyawadhwa makes sense! Update, PTAL 🙃

**Review by priyawadhwa — APPROVED** [review]

**Comment by priyawadhwa** [comment]
Thanks @concaf !

**Pull request merged**